chia/_tests/environments/wallet.py

```diff
@@ -247,9 +247,9 @@ async def change_balances(self, update_dictionary: Dict[Union[int, str], Dict[st
 
     async def wait_for_transactions_to_settle(
         self, full_node_api: FullNodeSimulator, _exclude_from_mempool_check: List[bytes32] = []
-    ) -> List[TransactionRecord]:
+    ) -> List[LightTransactionRecord]:
         # Gather all pending transactions
-        pending_txs: List[TransactionRecord] = await self.wallet_state_manager.tx_store.get_all_unconfirmed()
+        pending_txs: List[LightTransactionRecord] = await self.wallet_state_manager.tx_store.get_all_unconfirmed()
         # Filter clawback txs
         pending_txs = [
```

Comment:
It seems to me that asking for *all* unconfirmed here is the root of the issue. Did you explore asking only for specific unconfirmed transactions? e.g. have the database query filter for `CLAWBACK_INCOMING_TRANSACTION_TYPE`